Dockerfile

```diff
@@ -18,17 +18,19 @@ RUN apk add --no-cache --quiet \
 COPY package.json yarn.lock ./
 RUN yarn install --frozen-lockfile && yarn cache clean
 
-# Copy application code
-COPY . ./
+# Update file/directory permissions
+RUN chown -R deploy:deploy ./
 
-# Ensure COMMIT_HASH is present
-RUN test -n "$COMMIT_HASH" && \
-  echo $COMMIT_HASH > COMMIT_HASH.txt && \
-  # Update file/directory permissions
-  chown -R deploy:deploy ./
```

Comment:
Docker re-runs `chown -R...` upon every `hokusai dev start` which I had to do frequently, and it takes ~10min, excruciating. It seems that the `COMMIT_HASH` part busts docker cache. So I am breaking this step up: put `COMMIT_HASH` near the end and after `USER` deploy so the file is owned by deploy user, put `chown -R...` before COPY, so its cache is not busted by local file changes, add `--chown` flag to COPY to take care of files copied-in.